src/plugins/sockets/inGameEvents.js

```diff
@@ -125,57 +125,33 @@ export async function handleInGameEvents(evData) {
       break;
     case SocketEvent.REMATCH:
       gameStore.setRematch({ pNum: evData.pNum, rematch: evData.game[`p${evData.pNum}Rematch`] });
+      return;
+    case SocketEvent.NEW_GAME_FOR_REMATCH: {
+      // ignore if not currently in/spectating relevant game
       if (
-        evData.pNum === gameStore.myPNum &&
-        currentRoute.name !== ROUTE_NAME_SPECTATE &&
-        evData.game[`p${evData.pNum}Rematch`]
+        Number(urlGameId) !== evData.oldGameId || 
+        ![ROUTE_NAME_GAME, ROUTE_NAME_SPECTATE].includes(currentRoute.name)
       ) {
-        router.push({
-          name: ROUTE_NAME_REMATCH,
-          params: {
-            gameId: gameStore.id,
-          },
-        });
+        return;
       }
-      return;
-    case SocketEvent.NEW_GAME_FOR_REMATCH: {
       gameStore.setRematchGameId(evData.gameId);
+      const { gameId: oldGameId } = currentRoute.params;
 
       if (currentRoute.name === ROUTE_NAME_SPECTATE) {
+        await gameStore.requestSpectate(evData.gameId);
         gameStore.updateGame(evData.newGame);
```

Comment:
`requestSpectate()` calls `updateGame()` inside of its response handlers so this is probably uneeded